### PR TITLE
Implement job error recovery

### LINKED ISSUE #1370: Handling errors of evaluation job
Allow students to resubmit the answer when evaluation failed.

### PR DESCRIPTION
**Pull request by allenwq**
This PR fixes #1370:
Allows users to track the status of the job even after page refreshing.
Allows users to resubmit the job when there's a timeout or other errors in the evaluation job.

So once the job failed, they won't complain to us that the buttons are missing :)

### CODE REVIEW

File: app/assets/javascripts/course/assessment/submission/answer/programming_submit.js [hunk 1]
```diff
@@ -6,55 +6,69 @@
...
 
-  function submitFormAndWaitForJob($form, answer_id) {
+  // Find all the spinning jobs in the page and
+  function checkSubmittedJobs() {
+    $('a.btn.submitted').filter(DOCUMENT_SELECTOR + '*').each(function(index) {
+      var answer = $(this).closest('.answer');
+      var jobUrl = $(this).data('job-path');
+      var answerId = answer.data('answer-id');
+      // Use different delays, so that all the requests won't send at once.
+      waitForJob(jobUrl, answerId, (index + 1) * DELAY);
```
Comment (fonglh) [inline app/assets/javascripts/course/assessment/submission/answer/programming_submit.js]: The later answers will have longer delays?

Comment (allenwq) [inline app/assets/javascripts/course/assessment/submission/answer/programming_submit.js]: This is the delay for the initial request, later they are all the same.

Comment (fonglh) [inline app/assets/javascripts/course/assessment/submission/answer/programming_submit.js]: Ok I see it, will merge when it passes then.

**Comment by fonglh** [comment]
Was the earlier test failure random?

**Comment by allenwq** [comment]
It's a failure of cannot connect to lvh.me... which happens pretty frequently.

**Pull request merged**